tf2_ros/include/tf2_ros/transform_listener.h

```diff
@@ -84,14 +114,10 @@ class TransformListener
     bool spin_thread,
     const rclcpp::QoS & qos,
     const rclcpp::QoS & static_qos,
-    const rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> & options = []() {
-      rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> options;
-      options.qos_overriding_options = rclcpp::QosOverridingOptions{
-        rclcpp::QosPolicyKind::Depth,
-        rclcpp::QosPolicyKind::History,
-        rclcpp::QosPolicyKind::Reliability};
-      return options;
-    } ())
+    const rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> & options =
+    detail::get_default_transform_listener_sub_options<AllocatorT>(),
```

Comment:
But I still don't see why we need a default argument here (and below).  This is a private method, so we know for sure that the only callers are internal to this class, and we can make sure that they all send the correct arguments down.  It's a bit of a philosophical question (as it will work just fine this way), but I like to avoid default arguments for private methods.

Reply:
Done in https://github.com/ros2/geometry2/pull/383/commits/e8823c496bc4251ead583d89235e1255a854e4b1